lib/vechain/lib/assets/instance/node/userdata.sh

```diff
@@ -0,0 +1,259 @@
+  node_options+=("--network main")
+  snapshot_url="${snapshot_url}-mainnet"
+elif [ "_NETWORK_" == "testnet" ]; then
+  node_options+=("--network test")
+  snapshot_url="${snapshot_url}-testnet"
+fi
+
+snapshot_url="${snapshot_url}.tar.zst"
+
+# Get instance ID
+TOKEN=$(curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+INSTANCE_ID=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" -s http://169.254.169.254/latest/meta-data/instance-id)
+
+cat >> "$ENV_FILE" <<EOF
+export AWS_REGION=_AWS_REGION_
+export STACK_NAME=_STACK_NAME_
+export INSTANCE_ID=$INSTANCE_ID
+export RESOURCE_ID=_NODE_CF_LOGICAL_ID_
+export DATA_VOLUME_TYPE=_DATA_VOLUME_TYPE_
+export DATA_VOLUME_SIZE=_DATA_VOLUME_SIZE_
+export NETWORK=_NETWORK_
+export VET_NODE_TYPE=_VET_NODE_TYPE_
+export SYNC_FROM_PUBLIC_SNAPSHOT=_SYNC_FROM_PUBLIC_SNAPSHOT_
+export VET_CONTAINER_IMAGE=_VET_CONTAINER_IMAGE_
+export ASSETS_S3_PATH=_ASSETS_S3_PATH_
+export NODE_OPTIONS="${node_options[*]}"
+export COMPOSE_FILE=${docker_compose_files}
+export LIFECYCLE_HOOK_NAME=_LIFECYCLE_HOOK_NAME_
+export ASG_NAME=_ASG_NAME_
+export SNAPSHOT_URL=${snapshot_url}
+EOF
+
+source $ENV_FILE
+
+# Add environment variables to /etc/profile so they are available on login for root user
+cat >> /etc/profile <<EOF
+# Automatically source the environment file on login
+if [ "\$(id -u)" -eq 0 ] && [ -f $ENV_FILE ]; then
+  source $ENV_FILE
+fi
+EOF
+
+# Install Middleware
+sudo yum update -y
+sudo yum install jq docker wget cronie amazon-cloudwatch-agent zstd -y
+
+# Install Docker and setup docker compose
+# Add ec2-user to docker group and start Docker
+sudo usermod -a -G docker ec2-user
+sudo systemctl enable docker.service
+sudo systemctl start docker.service
+
+# Detect architecture
+arch=$(uname -m)
+
+# Create docker CLI plugins dir in user's home
+mkdir -p ~/.docker/cli-plugins
+
+case $arch in
+  x86_64)
+    echo "This server is using Intel/AMD (x86_64)."
+
+    # Docker Compose (x86_64)
+    wget -q https://github.com/docker/compose/releases/download/v2.38.2/docker-compose-linux-x86_64 \
+      -O ~/.docker/cli-plugins/docker-compose
+
+    # s5cmd (x86_64)
+    wget -q https://github.com/peak/s5cmd/releases/download/v2.2.2/s5cmd_2.2.2_Linux-64bit.tar.gz \
+      -O /tmp/s5cmd.tar.gz
+    ;;
+
+  aarch64|arm64)
+    echo "This server is using AWS Graviton (aarch64/arm64)."
+
+    # Docker Compose (ARM64)
+    wget -q https://github.com/docker/compose/releases/download/v2.38.2/docker-compose-linux-aarch64 \
+      -O ~/.docker/cli-plugins/docker-compose
+
+    # s5cmd (ARM64)
+    wget -q https://github.com/peak/s5cmd/releases/download/v2.2.2/s5cmd_2.2.2_Linux-arm64.tar.gz \
+      -O /tmp/s5cmd.tar.gz
+    ;;
+
+  *)
+    echo "Processor architecture is unknown or unsupported: $arch"
+    exit 1
+    ;;
+esac
+
+# Make docker-compose executable
+chmod +x ~/.docker/cli-plugins/docker-compose
+
+tar -xvf /tmp/s5cmd.zip -C /tmp && mv /tmp/s5cmd /usr/bin/
```

Comment:
Maybe I am missing something but we are creating a file with the extension `tar.gz` above, right? so `s5cmd.tar.gz` instead of `zip`

```suggestion
tar -xzf /tmp/s5cmd.tar.gz -C /tmp && mv /tmp/s5cmd /usr/bin/
```

Reply:
Great point, I missed this. S5CMD is not used at all in this implementation, this code segment can be removed.